lib/Role/utils.js

```diff
@@ -169,3 +171,30 @@ export function getCheckboxHandlers({
     isAllActionCapabilitySetsSelected,
   };
 }
+
+/**
+ * Calculates the number of items (capabilities or capability sets) that would be unselected
+ * when certain applications are unselected.
+ *
+ * @param {Array} unselectedAppIds - Array of application IDs that are being unselected.
+ * @param {Object} items - An object where keys are item categories and values are arrays of items.
+ * @param {Object} checkedMap - A map indicating which items are currently selected.
+ * 
+ * @returns {number} The count of items that would be unselected.
+ */
+export function getUnselectedItemCount(unselectedAppIds, items, checkedMap) {
+  let unselectedItemCount = 0;
+
+  for (const [key, itemList] of Object.entries(items)) {
+    for (const item of itemList) {
+
+      for (const appId of unselectedAppIds) {
+        if (item.applicationId === getApplicationName(appId) && checkedMap[item.id]) {
```

Comment:
Either a variable or a function is misnamed here: either `item.applicationId` actually contains an application _name_, or `getApplicationName(appId)` actually exchanges a name for an id. Please fix whichever one is wrong.

Reply:
Yes, this is confusing. I updated the function name to indicate we are needing to compare the unversioned application ID that is displayed on the UI. Also, provided comments to help clarify.